api/src/main/java/org/apache/iceberg/ExpireSnapshots.java

```diff
@@ -55,6 +55,15 @@
    */
   ExpireSnapshots expireOlderThan(long timestampMillis);
 
+  /**
+   * Retains at least last N snapshots and expires all snapshots older than the given timestamp
+   * in {@link #expireOlderThan(long)}.
```

Comment:
I would also mention (maybe in the implementation) that any in-flight snapshots during this operation will be added so the resulting number of snapshots can be more than `numSnapshots`

Reply:
@prodeezy sure, will do

Reply:
I added this to the docs: